lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -494,52 +495,50 @@ def post(self, request):
-                    self.msg += _(
-                        'Error - cannot get course with ID {0}<br/><pre>{1}</pre>'
-                    ).format(
-                        course_id,
-                        escape(str(err))
+                except Exception as err:
+                    self.msg += _('Error - cannot get course with ID {0}<br/><pre>{1}</pre>').format(
+                        course_id, escape(str(err))
                     )
 
         elif action == 'del_course':
             course_id = request.POST.get('course_id', '').strip()
-            course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-            course_found = False
-            if course_key in courses:
-                course_found = True
-                course = courses[course_key]
+            try:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+            except InvalidKeyError as e:
+                self.msg += _('Field Course ID or dir filled incorrectly')
             else:
-                try:
-                    course = get_course_by_id(course_key)
+                course_found = False
+                if course_key in courses:
                     course_found = True
-                except Exception, err:   # pylint: disable=broad-except
-                    self.msg += _(
-                        'Error - cannot get course with ID {0}<br/><pre>{1}</pre>'
-                    ).format(
-                        course_key,
-                        escape(str(err))
-                    )
-
-            if course_found:
-                # delete course that is stored with mongodb backend
-                self.def_ms.delete_course(course.id, request.user.id)
-
-                # delete search index
-                try:
-                    response = self._searcher.search(doc_type="courseware_content", field_dictionary={'course': course_id})
-                    result_ids = [result["data"]["id"] for result in response["results"]]
-                    self._searcher.remove('courseware_content', result_ids)
-                    self._searcher.remove('course_info', [course_id])
-                except Exception as e:  # pragma: no cover
-                    log.error(e.message)
-
-                CourseOverview.objects.filter(id=course.id).delete()
-
-                # don't delete user permission groups, though
-                self.msg += \
-                    u"<font color='red'>{0} {1} = {2} ({3})</font>".format(
-                        _('Deleted'), course.location.to_deprecated_string(), course.id.to_deprecated_string(), course.display_name)
+                    course = courses[course_key]
+                else:
+                    try:
+                        course = get_course_by_id(course_key)
+                        course_found = True
+                    except Http404 as err:   # pylint: disable=broad-except
```

Comment:
Pylint pragma command is no longer needed, since we're now using a specific exception.
```suggestion
                    except Http404 as err:
```